freqtrade/freqtradebot.py

```diff
@@ -192,7 +194,28 @@ def process(self) -> None:
 
         strategy_safe_wrapper(self.strategy.bot_loop_start, supress_error=True)()
 
-        self.strategy.analyze(self.active_pair_whitelist)
+        if self.external_signal_controller:
+            if not self.external_signal_controller.is_leader():
+                # Run Follower mode analyzing
+                leader_pairs = self.pairlists._whitelist
+                self.strategy.analyze_external(self.active_pair_whitelist, leader_pairs)
+            else:
+                # We are leader, make sure to pass callback func to emit data
+                def emit_on_finish(pair, dataframe, timeframe, candle_type):
```

Comment:
So again, going with a comment above, having the `Strategy` class itself take care of this would mean coupling the `ExternalSignalController` with the `IStrategy` object. I think that would be bad practice as the `FreqTradeBot` object is the main part where everything is essentially "glued" together. So having that manage the  external signals just made sense as the "flow" of calculations is slightly different depending on being a Follower or normal instance.